File: 304/A.py
N = int(input())
SA = []
min_idx = 0
for n in range(N):
    S, A = [l for l in input().split()]
    SA.append([S, int(A)])
    if SA[n][1] < SA[min_idx][1]:
        min_idx = n

for n in range(N):
    print(SA[(min_idx + n) % N][0])

File: 304/A.c
int main() {
    int N, idx, min_idx = 0;
    int A[100];
    char S[100][10];
    scanf("%d", &N);
    for (idx = 0; idx < N; idx++) {
        scanf("%s %d", S[idx], &A[idx]);
        if (A[idx] < A[min_idx])min_idx = idx;
    }
    for (idx = 0; idx < N; idx++) {
        printf("%s\n", S[(idx + min_idx) % N]);
    }
    return 0;
}
